modules/home-manager/firefox.nix

```
@@ -42,12 +58,46 @@ let
                   accentSupport = true;
                 }
                 // {
-                  force = mkOption {
-                    type = types.bool;
-                    default = false;
-                    description = "Forcibly override any existing configuration for Firefox Color.";
-                    example = true;
-                  };
+                  force = mkEnableOptionDesc "Forcibly override any existing configuration for Firefox Color.";
+
+                  userstyles =
+                    catppuccinLib.mkCatppuccinOption {
+                      name = "${name} userstyles";
+                      accentSupport = true;
+                    }
+                    // {
+                      force = mkEnableOptionDesc "Forcibly override any existing configuration for Stylus.";
+
+                      # Since applying settings through home-manager makes the storage.js file read only,
+                      # build-stylus-settings removes the ability to configure userstyles through the stylus extension,
+                      # instead settings are defined here & are baked into the userstyles
+                      settings = mkOption {
+                        type = types.attrsOf (
+                          types.attrsOf (
+                            types.oneOf [
+                              types.str
+                              types.int
+                              types.bool
+                            ]
+                          )
+                        );
+                        default = { };
+                        description = "Settings for each userstyle.";
+                        example = {
+                          youtube = {
+                            lightFlavor = "latte";
+                            darkFlavor = "mocha";
+                            accentColor = "mauve";
+
+                            logo = true;
+                            oled = false;
+                            sponsorBlock = true;
+                          };
+
+                          github.enable = false;
+                        };
```

this should really be put through [lib.literalExample](https://noogle.dev/f/lib/literalExample)

```suggestion
                        example = lib.literalExample ''
                          {
                          	youtube = {
                           	  lightFlavor = "latte";
                              darkFlavor = "mocha";
                              accentColor = "mauve";

                              logo = true;
                              oled = false;
                              sponsorBlock = true;
                            };

                            github.enable = false;
                          };
                        '';
```